assets/dev/js/editor/editor-base.js

```diff
@@ -1531,4 +1551,49 @@ export default class EditorBase extends Marionette.Application {
 
 		$files.attr( { type } );
 	}
+
+	getElementType( elType, widgetType = false ) {
+		let index = elType,
+			result = this.registeredElements[ index ];
+
+		if ( 'widget' === elType ) {
+			if ( ! widgetType ) {
+				throw new TypeError( 'Missing widget type' );
+			}
+
+			index += '-' + widgetType;
+		}
+
+		if ( this.registeredElements[ index ] ) {
+			result = this.registeredElements[ index ];
+		}
+
+		return result;
+	}
+
+	/**
+	 * @param {ElementBase} element
+	 */
+	registerElementType( element ) {
+		// Validate instanceOf `element`.
+		if ( ! ( element instanceof ElementBase ) ) {
+			throw new TypeError( 'The element argument must be an instance of ElementBase.' );
+		}
+
+		const index = element.getTypeIndex();
+
+		if ( this.registeredElements[ index ] ) {
+			throw new Error( 'Element type already registered' );
+		}
+
+		this.registeredElements[ index ] = element;
+	}
+
+	registerElements() {
+		Object.values( elements ).forEach( ( ElementClass ) => {
```

Comment:
Shouldn't be `this.elements` or someting?

Reply:
no, it will be saved in `this.registeredElements`, `elements` is index of classes to be created.